tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/VersionConflict/VersionConflict2xTests.cs

```diff
@@ -73,7 +73,10 @@ public void SubmitTraces()
 #endif
 
                 // Check the headers of the outbound http requests
-                var outputLines = processResult.StandardOutput.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+                var outputLines = processResult.StandardOutput
+                                               .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                                               .Where(line => char.IsDigit(line[0])) // filter out unrelated profiler log lines
```

Comment:
Which extra log lines do we have here? 🤔 Presumably the number of log lines is deterministic, if so, can we update the expected count instead? Otherwise we may risk filtering out something we don't want to?

Reply:
The log lines that are polluting the console output are the ones from the Native Loader, which currently isn't utilizing `spdlog` at all, but instead [writing to standard output directly](https://github.com/DataDog/dd-trace-dotnet/blob/a72b74fc8e67d8d8a6430628fe8643b3a693d2bc/tracer/src/Datadog.AutoInstrumentation.NativeLoader/logging.h#L30). I think this should definitely be fixed but that's probably out of scope for this PR, hence this workaround, but there is a bigger underlying question here:

Do we want our logs (both for the managed and native components of the client libraries, and for the native loader) to be written out to standard output? Personally, I'd argue the answer is "yes, if and only if we are running in `DEBUG` (local development scenario), or as part of an integration test". My reasoning for that is developer productivity - it would be quite handy if the output of every failing test in Azure DevOps would contain the full log output for the sample application that was executed.

If you agree with that idea, then we should probably keep this solution as-is?

Reply:
That's a fair point. While I think it'd be a nice productivity boost if we could see the contents of the relevant logs in the Attachments part of Azure DevOps's test explorer for every failing test, we can accomplish that in other ways (e.g. by explicitly appending them to xUnit's `ITestOutputHelper`). But that's a separate discussion.

I reverted this change in https://github.com/DataDog/dd-trace-dotnet/pull/2358/commits/923a1472e0033750113d4125ab34bbb08090990e and submitted #2481 which solves the issue this was trying to work around.